### PR TITLE
fix: handle insufficient input amount error in v2 add liquidity

### LINKED ISSUE #1520: Crash report: `t`
**Bug Description**
  
App crashed

**Steps to Reproduce**

1. I think I tried to create a liquidity position with a very small amount of a token in v2. 
2. Create new liquidity position (pool) by navigating in browser to v2 liquidity 
3. select ETH as prime token, and ER20 token as secondary 
4. Select max amount on the ER20 token
   ...
   
**Error**

```
t
```

**Stacktrace**

```
t
    at e.t.getLiquidityMinted (https://app.uniswap.org/static/js/4.7be9d8b6.chunk.js:2:88268)
    at https://app.uniswap.org/static/js/main.1cc75114.chunk.js:1:522644
    at Object.Eo [as useMemo] (https://app.uniswap.org/static/js/4.7be9d8b6.chunk.js:2:1477347)
    at t.useMemo (https://app.uniswap.org/static/js/4.7be9d8b6.chunk.js:2:1396801)
    at https://app.uniswap.org/static/js/main.1cc75114.chunk.js:1:522535
    at oS (https://app.uniswap.org/static/js/main.1cc75114.chunk.js:1:523520)
    at no (https://app.uniswap.org/static/js/4.7be9d8b6.chunk.js:2:1472614)
    at zo (https://app.uniswap.org/static/js/4.7be9d8b6.chunk.js:2:1482181)
    at Vs (https://app.uniswap.org/static/js/4.7be9d8b6.chunk.js:2:1526063)
    at Au (https://app.uniswap.org/static/js/4.7be9d8b6.chunk.js:2:1512246)
```

**Device data**

```json5
{
  "ua": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.93 Safari/537.36",
  "browser": {
    "name": "Chrome",
    "version": "90.0.4430.93",
    "major": "90"
  },
  "engine": {
    "name": "Blink",
    "version": "90.0.4430.93"
  },
  "os": {
    "name": "Mac OS",
    "version": "10.13.6"
  },
  "device": {},
  "cpu": {}
}
```

### PR DESCRIPTION
**Pull request by justindomingue**
fixes #1520

### CODE REVIEW

File: src/state/mint/hooks.ts [hunk 1]
```diff
@@ -160,10 +160,16 @@ export function useDerivedMintInfo(
       wrappedCurrencyAmount(currencyBAmount, chainId),
     ]
     if (pair && totalSupply && tokenAmountA && tokenAmountB) {
-      return pair.getLiquidityMinted(totalSupply, tokenAmountA, tokenAmountB)
-    } else {
-      return undefined
+      try {
+        return pair.getLiquidityMinted(totalSupply, tokenAmountA, tokenAmountB)
+      } catch (e) {
+        if (e instanceof InsufficientInputAmountError) {
```
Comment (moodysalem) [inline src/state/mint/hooks.ts:166]: idt instanceof works correctly with errors in typescript (i ran into issues with it before). can you repro and this fixes it?

Comment (moodysalem) [reply src/state/mint/hooks.ts:166]: ah use `isInsufficientInputAmountError`

Comment (justindomingue) [reply src/state/mint/hooks.ts]: instanceof was working for me but switched to isIns..

**Review by moodysalem — APPROVED** [review]

File: src/state/mint/hooks.ts [hunk 2]
```diff
@@ -160,10 +160,16 @@ export function useDerivedMintInfo(
       wrappedCurrencyAmount(currencyBAmount, chainId),
     ]
     if (pair && totalSupply && tokenAmountA && tokenAmountB) {
-      return pair.getLiquidityMinted(totalSupply, tokenAmountA, tokenAmountB)
-    } else {
-      return undefined
+      try {
+        return pair.getLiquidityMinted(totalSupply, tokenAmountA, tokenAmountB)
+      } catch (error) {
+        if (error.InsufficientInputAmountError) {
```
Comment (moodysalem) [inline src/state/mint/hooks.ts:166]: this is a typo. should be isInsufficientInputAmount

Comment (justindomingue) [reply src/state/mint/hooks.ts:166]: good catch ,submitted a pr

**Pull request merged**